builtin/providers/aws/resource_aws_iam_instance_profile.go

```diff
@@ -49,12 +55,33 @@ func resourceAwsIamInstanceProfile() *schema.Resource {
 					return
 				},
 			},
+
+			"name_prefix": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					// https://github.com/boto/botocore/blob/2485f5c/botocore/data/iam/2010-05-08/service-2.json#L8196-L8201
+					value := v.(string)
+					if len(value) > 64 {
```

Comment:
This value probably isn't correct... are there any guarantees made about the length of the string returned by the `uuidV4` function?